docs/enhancements/support-node-labeling.md

```diff
@@ -0,0 +1,159 @@
+---
+title: node labeling support
+authors:
+- "@oamizur"
+  creation-date: 2022-12-20
+  last-updated: 2023-01-30
+---
+
+# Support node labeling during installation
+
+## Summary
+
+There was a requirement to support node roles during installation.  These roles 
+are extra roles, which are used by the user and not by any Openshift components.
+These labels should be applied on a node in a spoke cluster.
+
+Although the requirement is specifically for roles, the implementation should support setting 
+any label on a node in a spoke cluster, regardless if the label represents a role.
+
+## Motivation
+
+Currently, nodes can be assigned with a single role which can be one of two values [ **master**, **worker**].
+The may want to add additional node labels that can be used as custom roles.  
+
+Each role in OCP are associated with machine config pool (MCP).  When a node reboots itself
+from the pointer ignition, it uses specific MCP that contains the specific configuration
+for the role.  This MCP has the format `rendered-<role>-*` (* stands for a hash string).  So there is MCP for master 
+role `rendered-master-*` and for worker `rendered-worker-*`.
+
+When adding custom role, it can be associated with custom MCP which references specific configuration
+for that role.  Setting appropriate label a node, causes the node to reboot and apply the 
+MCP configuration (machine config) on that node.
+
+Setting node labels should support this flow.
+
+### Goals
+- For ZTP flow: add capability to add and update annotations on BMH which will be transferred 
+to agent CR.
+- For boot it yourself flow: add capability to add and update labels to agent CR after agent
+  has been created.
+- Transfer the labels from the agent CR to the corresponding node on the spoke cluster.
+
+### Non goals
+- Single way to set node labels before agent creation both for ZTP flow, and boot it yourself flow.
+
+## Proposal
+
+### ZTP flow
+Labels can be applied on every CR.  So there is no known limitation to add a label on agent.
+
+For the ZTP flow, there should a pattern for transferring an annotation from the BMH to a label  
+in the corresponding agent.  The pattern is actually the prefix of the annotation.
+It should be **node-label**.  The suffix of the annotation will be transferred 
+to labels on the corresponding agent CR.
```

Comment:
This section contradicts the next section ("Node labels on an agent CR").  Here you say that we are "transferring an annotation from the BMH to a label in the corresponding agent" but below you say that annotations "will be transferred to the dedicated field nodeLabels".

I think the next section is correct and you can erase this one.  We should not automatically copy labels between resources but rather have it explicit in the API.

Reply:
OK